src/getters/crunchbase.py

```diff
@@ -0,0 +1,114 @@
+"""
+discovery_utils.src.getters.crunchbase.py
+
+Module for easy access to downloaded CB data on S3.
+
+"""
+import pandas as pd
+import boto3
+import logging
+import re
+
+
+# Define global variables
+BUCKET_NAME = "discovery-iss"
+S3_PATH = "data/crunchbase/"
+AWS_ACCESS_KEY =
+AWS_SECRET_KEY = 
+FILE_NAMES = [
+    "acquisitions.csv",
+    "category_groups.csv",
+    "checksum.csv",
+    "degrees.csv",
+    "event_appearances.csv",
+    "events.csv",
+    "funding_rounds.csv",
+    "funds.csv",
+    "investment_partners.csv",
+    "investments.csv",
+    "investors.csv",
+    "ipos.csv",
+    "jobs.csv",
+    "org_parents.csv",
+    "organization_descriptions.csv",
+    "organizations.csv",
+    "people_descriptions.csv",
+    "people.csv"
+]
+
+
+# Initialize S3 client
+S3 = boto3.client(
+    "s3", aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY
+    )
+
+
+# Building block functions - not to be used directly
+
+def list_s3_directories() -> list[str]:
+    """List S3 directories under the specified prefix."""
+    # List all objects with the specified prefix
+    objects = S3.list_objects_v2(Bucket=BUCKET_NAME, Prefix=S3_PATH, Delimiter='/')
+
+    # Extract the directory names from the CommonPrefixes list
+    directories = [prefix['Prefix'] for prefix in objects.get('CommonPrefixes', [])]
+
+    return directories
+
+def extract_timestamp(directory: str) -> str or None:
+    """ Use regular expression to extract the timestamp from a directory"""
+    match = re.search(r'\d{8}_\d{6}', directory)
+    if match:
+        timestamp_str = match.group()
+        return timestamp_str
+    else:
+        return None
+    
+def directory(position: int) -> str:
+    """Directory name for a given snapshot of CB data."""
+    
+    # Get the list of directories
+    directories = list_s3_directories()
+    
+    # Get a list of timestamps
+    timestamps = []
+    for directory in directories:
+        timestamps.append(extract_timestamp(directory))
+    
+    # Sort the timestamps in descending order
+    sorted_timestamps = sorted(timestamps, reverse=True)
+    
+    # Return the most recent directory
+    for directory in directories:
+        if sorted_timestamps[position] in directory:
+            return directory
```

Comment:
```suggestion
    return sorted_timestamps[position]
```
items in sorted_timestamps will only exist if it's in directories

Reply:
Hey @sqr00t, thanks for the suggestion. I want to return the directory, not the timestamp, so this needs to end `return directory`. Unless I'm missing something, I think this means these lines are fine? I want to take each directory one by one, and return the directory which contains the chosen timestamp.

Reply:
this actually links with the global sorted directories list I mentioned above